runner.sh

```diff
@@ -0,0 +1,169 @@
+            ====================================
+            ILLUMINA ANALYSIS COMPLETED!!!
+            ===================================="
+      
+      echo "
+            =======================================
+            DELETING SAMPLE DIRECTORY $ILLUM_RES 
+            ======================================="
+
+      mkdir $ILLUM_RES
+      cp -r $PWD/data/illumina/$ILLUM_RES $PWD/$ILLUM_RES
+      wait
+      rm -rf $PWD/data/illumina/$ILLUM_RES
+fi
+
+
+elif [ $OUT_DIRS == "artic" ]; then
+echo "
+     =================================================
+     FOUND NANOPORE ARTIC SAMPLE DIRECTORY: $ARTIC_RES 
+     ================================================="
+echo "
+     ========================================== 
+     CHECK ARTIC METADATA FILE: $ARTIC_RES  
+     ==========================================="
+
+python3 sampleValidation.py $PWD/data/artic/$ARTIC_RES/ >check_samples.out
+wait
+python3 cli.py -i $PWD/data/artic/$ARTIC_RES/*.csv -s rc/definition.yaml >metadata_check.out
+wait
+
+echo "
+     ==================================================== 
+     FINISHED CHECKING ARTIC METADATA FILE: $ARTIC_RES
+     ===================================================="
+
+if grep -F "CHECK" check_samples.out || grep -F "not" metadata_check.out
+then
+      echo "Please check your input files." 
+else
+      echo "
+            ===========================================
+            START ANALYSIS WITH NANOPORE ARTIC PIPELINE 
+            ==========================================="
+            nextflow run main.nf -profile singularity \
+                  --nanopolish --prefix "${ARTIC_RES}_nanopore" \
+                  --basecalled_fastq data/artic/${ARTIC_RES}/fastq_pass/ \
+                  --fast5_pass data/artic/${ARTIC_RES}/fast5_pass/ \
+                  --sequencing_summary data/artic/${ARTIC_RES}/*.txt \
+                  --outdir ${ARTIC_RES}_Results_`date +%Y-%m-%d`
+      wait
+      echo "
+            ======================================
+            NANOPORE ARTIC ANALYSIS COMPLETED!!!  
+            ======================================"
+      echo "
+            ====================================
+            DELETING SAMPLE DIRECTORY $ARTIC_RES 
+            ===================================="
+
+      mkdir $ARTIC_RES
+      cp -r $PWD/data/artic/$ARTIC_RES $PWD/$ARTIC_RES
+      wait
+      rm -rf $PWD/data/artic/$ARTIC_RES
+fi
+
+else [[ $OUT_DIRS == "midnight" ]]
+echo "
+     ==============================================
+     FOUND NANOPORE SAMPLE DIRECTORY: $MIDNIGHT_RES 
+     =============================================="
+echo "
+     =========================================== 
+     CHECK MIDNIGHT METADATA FILE: $MIDNIGHT_RES
+     ==========================================="
+
+python3 sampleValidation.py $PWD/data/midnight/$MIDNIGHT_RES/ >check_samples.out
+wait
+python3 cli.py -i $PWD/data/midnight/$MIDNIGHT_RES/*.csv -s rc/definition.yaml >metadata_check.out
```

Comment:
This invocation(s) look a bit yank. Consider invoking the cli through the package, and setting output path as an actual option as opposed to piping

Reply:
yes, this is actually to guide the nextflow run to find the paths. I will check this as you suggested.

Reply:
I removed this part of the code as HCPInterface is doing this job; I don't need to include this on the main script. :)